### PR TITLE
dbt: DBT job trigger in warehouse materializations

### PR DESCRIPTION
**Pull request by mdibaiee**
**Description:**

- Trigger DBT jobs after transactions which materialize data

**Workflow steps:**

(How does one use this feature, and how has it changed)

**Documentation links affected:**

(list any [documentation links](https://docs.google.com/document/d/1SRC9VS9zyCzWl3n4HXHbc4wPB1eLxJHkA2rtu9ZNokM/edit?usp=sharing) that you created, or existing ones that you've identified as needing updates, along with a brief description)

**Notes for reviewers:**

(anything that might help someone review this PR)

- - -
This change is [Reviewable](https://reviewable.io/reviews/estuary/connectors/1829)

### CODE REVIEW

**Comment by dyaffe** [comment]
This option only existing if a sync schedule is specified makes sense to me.

Also, would it be possible to have our first version include all warehouse?  That would involve adding support in Snowflake, Bigquery and Redshift.

**Comment by mdibaiee** [comment]
@dyaffe just had a chat with Dani regarding the main concern, now going to add an option to choose between three different modes of treating already-running jobs: `skip` triggering a job if one is already running, `replace` the running job with a new one, or `ignore` the running job and trigger the job anyway.

After that will add to all warehouses

File: materialize-databricks/.snapshots/TestSpecification [hunk 1]
```diff
@@ -138,7 +189,8 @@
         "type": "object",
         "title": "Advanced Options",
         "description": "Options for advanced users. You should not typically need to modify these.",
-        "advanced": true
+        "advanced": true,
+        "order": 6
```
Comment (williamhbaker) [inline materialize-databricks/.snapshots/TestSpecification:193]: A few of these have `order` set in their configurations but others don't - just wondering if there is a specific reason for that?

**Review by williamhbaker — APPROVED** [review]
LGTM

At some point in the future it could be worth considering moving some of this behavior to the transactor, like we have done for the sync schedule. Right now it's only for SQL materializations but I could see others (Iceberg) also wanting to use this which would make it interesting as a general feature there.

File: materialize-databricks/.snapshots/TestSpecification (hunk 1 above)
Comment (mdibaiee) [reply materialize-databricks/.snapshots/TestSpecification:193]: I haven't set order for `dbt_job_trigger` at all, some connectors' configurations generally have `order` while others don't... the diff is for other properties of those connectors

**Pull request merged**